src/components/radio-button-group/radio-button-group.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+    Component,
+    Event,
+    EventEmitter,
+    h,
+    Prop,
+    State,
+    Watch,
+} from '@stencil/core';
+import { ListItem } from '../list/list-item.types';
+import { LimelListCustomEvent } from '@limetech/lime-elements';
+
+/**
+ * Radio Button Group
+ *
+ * The Radio Button component provides a convenient way to create a group of radio buttons
+ * from an array of options. Radio buttons allow users to select a single option from
+ * multiple choices, making them ideal for exclusive selections.
+ *
+ * :::note
+ * A single radio button is never useful in a UI. Radio buttons should always come in groups
+ * of at least 2 options where only one can be selected at a time.
+ * :::
+ *
+ * @exampleComponent limel-example-radio-button-group-basic
+ * @exampleComponent limel-example-radio-button-group-icons
+ * @exampleComponent limel-example-radio-button-group-multiple-lines
+ * @beta
+ */
+@Component({
+    tag: 'limel-radio-button-group',
+    shadow: false,
+})
+export class RadioButtonGroup {
+    /**
+     * Array of radio button options to display
+     */
+    @Prop()
+    public options: Array<ListItem> = [];
+
+    /**
+     * The currently selected value
+     */
+    @Prop()
+    public value: any;
+
+    /**
+     * Disables all radio buttons when `true`
+     */
+    @Prop({ reflect: true })
+    public disabled = false;
+
+    /**
+     * Set to `true` if the radio button group should display larger icons with a background
+     */
+    @Prop({ reflect: true })
+    public badgeIcons: boolean;
```

Comment:
I don't like this name, maybe would have been good to name it like `hasBadgeIcons` or something like that since it is a boolean. I saw this name in many places even in the interface `ListRendererConfig`, so we keep it like this. But I just thought that was not a name we chose from the beginning.

Reply:
Yes, I totally agree with you. I just copied form existing naming. Since we're making just a wrapper for `limel-list`, it felt we could use the same name. But personally, I have no problems changing it.